lib/srv/desktop/rdp/rdpclient/client.go

```diff
@@ -305,14 +310,37 @@ func (c *Client) startRustRDP(ctx context.Context) error {
 			allow_directory_sharing: C.bool(c.cfg.AllowDirectorySharing),
 			show_desktop_wallpaper:  C.bool(c.cfg.ShowDesktopWallpaper),
 		},
-	); res.err_code != C.ErrCodeSuccess {
-		if res.message == nil {
-			return trace.Errorf("unknown error: %v", res.err_code)
-		}
+	)
+
+	var message string
+	if res.message != nil {
+		message = C.GoString(res.message)
 		defer C.free_string(res.message)
-		return trace.Errorf("%s", C.GoString(res.message))
 	}
 
+	// If the client exited with an error, send a tdp error notification and return it.
+	if res.err_code != C.ErrCodeSuccess {
+		var err error
+
+		if message != "" {
+			err = trace.Errorf("RDP client exited with an error: %v", message)
+		} else {
+			err = trace.Errorf("RDP client exited with an unknown error")
+		}
+
+		c.sendTDPNotification(err.Error(), tdp.SeverityError)
+		return err
+	}
+
+	if message != "" {
+		message = fmt.Sprintf("RDP client exited gracefully with message: %v", message)
```

Comment:
This is better than what we had, but still not super user friendly:
<img width="565" alt="image" src="https://github.com/gravitational/teleport/assets/7527103/6083a7fe-9536-4eb4-b2e7-7acdb8eafb43">

With rdp-rs we had the lookup table with human readable messages for each disconnect code: 

https://github.com/gravitational/rdp-rs/blob/edfb5330a11d11eaf36d65e4300555368b4c6b02/src/core/global.rs#L309-L339

Doesn't need to be addressed in this PR but I think we should get it done soon.

Reply:
https://github.com/gravitational/teleport/issues/38373